applications/callflow/src/module/cf_lookupcidname.erl

```diff
@@ -0,0 +1,120 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2015, 2600Hz INC
+%%% @doc
+%%% This module looks up the Caller ID Name by matching
+%%% numbers/patters with the provided lists.
+%%%
+%%% "data":{
+%%%   "lists": ["01fc63f92d9b89a25dd4ff1039e64497"]
+%%% },
+%%% @end
+%%% @contributors
+%%%     Sponsored by Conversant Ltd,
+%%%         implemented by SIPLABS, LLC (Ilya Ashchepkov)
+%%%-------------------------------------------------------------------
+-module(cf_lookupcidname).
+
+-export([handle/2]).
+
+-include("../callflow.hrl").
+
+-spec handle(wh_json:object(), whapps_call:call()) -> 'ok'.
+handle(Data, Call) ->
+    CallerIdNumber = whapps_call:caller_id_number(Call),
+    ListEntries = wh_json:get_value(<<"lists">>, Data),
+    AccountDb = whapps_call:account_db(Call),
+    lager:debug("matching ~p in ~p", [CallerIdNumber, AccountDb]),
+    CallerName = case match_prefix_in_lists(AccountDb, CallerIdNumber, ListEntries) of
+                     'continue' ->
+                         lager:debug("matching regexps"),
+                         match_regexp_in_lists(AccountDb, CallerIdNumber, ListEntries);
+                     {'stop', CallerName_} -> CallerName_
+                 end,
+    case CallerName of
+        'undefined' -> cf_exe:continue(Call);
+        CallerName ->
+            lager:info("setting caller name to ~p", [CallerName]),
+            cf_exe:continue(whapps_call:set_caller_id_name(CallerName, Call))
+    end.
+
+-spec match_prefix_in_lists(ne_binary(), ne_binary(), ne_binary() | [ne_binary()]) ->
+    {'stop', api_binary()} | 'continue'.
```

Comment:
@fenollp, could you write 40-41 lines with correct indention?
Just want to know to correct my editor.